### PR TITLE
Feature/mailchimp

### PR DESCRIPTION
**Pull request by hobadams**
Adding a simple mailchimp form to the footer of the docs. I've added a 'referrer' field so we can segment these users and send them release notes (I hope).

Screenshot 2020-06-19 at 11 25 03

### CODE REVIEW

File: src/components/Newsletter/Mailchimp.js
Comment (nimobeeren) [inline src/components/Newsletter/Mailchimp.js:46]: What does the `jsonp` call do here? Is it required by Mailchimp?

File: src/components/Newsletter/index.js
Comment (nimobeeren) [inline src/components/Newsletter/index.js:14]: Is that the right place to put the submit URL? I'm not sure what `themConfig` is normally used for.

**Comment by nimobeeren** [comment]
I tested the deploy preview, but signing up didn't quite work for me. The signup button became disabled when I clicked it, and I didn't see any network requests happening.

File: src/components/Newsletter/index.js
Comment (hobadams) [reply src/components/Newsletter/index.js:14]: The theme config is used API keys, analytics tags etc. This seems to be the docusaurous convention.

File: src/components/Newsletter/Mailchimp.js
Comment (hobadams) [reply src/components/Newsletter/Mailchimp.js:46]: @nimobeeren - I took inspiration from this [package](https://github.com/gndx/react-mailchimp-form/blob/master/src/index.js). I was going to use the package but it didn't quite do what I wanted. I'll check there a no security concerns etc using this method....if not it makes sense to keep it this way. I'll let you know

Comment (hobadams) [reply src/components/Newsletter/Mailchimp.js:46]: `jsonp` is the format Mailchimp expects and this seems fine. Ideally we'd use the API but for the sake of simplicity I'm happy with this solution.

**Comment by nimobeeren** [comment]
Just repeating what we discussed in Slack convo: it seems like Firefox likes to block the request for signing up to the newsletter. Would be great if we could do something about that.

**Comment by Borales** [comment]
@nimobeeren I remember @EddyVinck was having the same problem.. @EddyVinck did you solve it somehow?

**Comment by EddyVinck** [comment]
For mailchimp I used an extension that was originally written for PushPro. I've adjusted it for Falcon 2 here: https://gitlab.com/deity-io/deity-corporate-website-2/-/blob/feature/bc-app-pages/server/src/falcon-mailchimp-api/index.js

and you can see the usage in a  form here: https://gitlab.com/deity-io/deity-corporate-website-2/-/blob/feature/bc-app-pages/client/src/components/forms/Newsletter/NewsletterForm.js

I just pushed a client-side fix which should be available in a few moments on the test website (which I won't link here since this repo is public)

image

**Comment by EddyVinck** [comment]
I just did some digging regarding Firefox related problems with deity.io.

I found 1 commit that references a problem with Firefox from January 21st 2020 but it was regarding Mandrill and at that time it was still a client-side implementation. Eventually I moved the Mandrill logic to the server and that fixed that problem. 

I do not know if we had these problems with Mailchimp or not, but I don't think so as that implementation has always been server-side as that extension was already written for PushPro. 

I assume this problem with Mailchimp is similar to the situation I was having with Mandrill and that it has to do with the browser blocking something for security reasons.

**Comment by nimobeeren** [comment]
@EddyVinck indeed it seems like Firefox is blocking the request. However, we're using JSONP here which should bypass any CORS restrictions. Hard to say exactly why it's being blocked. Unfortunately moving to the server is not possible here since its a static site.

**Comment by hobadams** [comment]
@nimobeeren @EddyVinck - from reading a good amount of posts with others experiencing this issue I don't think there is a way around it. A few people even suggested that Mailchimp is on Firefox's block list?!? - although that sounds unlikely my thoughts are that it's not worth spending loads more time on it. What do you think?

**Comment by nimobeeren** [comment]
@hobadams that's weird. I guess the browser doesn't detect that the request is made through user action and not as a random tracking script. Anyway, I heard we have a nice product that can help us notify people who are interested! No need to bother with ancient email! :smile:

**Comment by hobadams** [comment]
> @hobadams that's weird. I guess the browser doesn't detect that the request is made through user action and not as a random tracking script. Anyway, I heard we have a nice product that can help us notify people who are interested! No need to bother with ancient email! 😄

@nimobeeren indeed!!! There is a task for adding push pro....just waiting on a core PR to be merged that adds a service worker and offline support.

**Comment by EddyVinck** [comment]
Is it not possible to have this static site send a request to a server? Maybe we could add an endpoint to deity.io once the BC app stuff is merged in. Or maybe even a serverless function/lambda?

**Comment by nimobeeren** [comment]
Doing this with serverless could work well, we're already deploying through Netlify so it should be relatively easy to set up. Let's leave that out of this PR for now though :D

**Comment by hobadams** [comment]
@EddyVinck @Borales - what are your thoughts on merging this?

**Comment by Borales** [comment]
@hobadams how about using https://www.npmjs.com/package/axios-hooks for query hooks?

**Review by Borales — APPROVED** [review]

**Pull request merged**
